src/ansys/fluent/core/quantity/quantity.py

```diff
@@ -0,0 +1,332 @@
+import ansys.fluent.core.quantity as q
+
+
+class Quantity(float):
+    """Quantity instantiates physical quantities using their real values and
+    units. All the instances of this class are converted to base SI units system to have
+    consistency in arithmetic operations.
+
+    Parameters
+    ----------
+    value : int | float
+        Real value of quantity.
+    units : str
+        Unit string representation of quantity.
+    quantity_map : dict
+        Quantity map representation of quantity.
+    dimensions : list
+        Dimensions representation of quantity.
+
+    Methods
+    -------
+    to()
+        Converts to given unit string.
+    convert()
+        Converts to given unit system.
+
+    Returns
+    -------
+    Quantity instance.
+    """
+
+    def __new__(cls, value, units=None, quantity_map=None, dimensions=None):
+        if (
+            (units and quantity_map)
+            or (units and dimensions)
+            or (quantity_map and dimensions)
+        ):
+            raise ValueError(
+                "Quantity only accepts 1 of the following: units, quantity_map, dimensions"
+            )
+
+        _units_table = q.UnitsTable()
+        _value = float(value)
+
+        if units or units == "":
+            _unit = units
+
+        if quantity_map:
+            units = q.QuantityMap(quantity_map).units
+            _unit = units
+
+        if dimensions:
+            _dimensions = q.Dimensions(dimensions=dimensions)
+            _unit = _dimensions.units
+
+        _type = _units_table.get_type(_unit)
+        _, si_multiplier, si_offset = _units_table.si_data(units=_unit)
+        _si_value = (
+            ((_value + si_offset) * si_multiplier)
+            if _type == "Temperature"
+            else (_value * si_multiplier)
```

Comment:
The call to `_units_table.si_data(units=_unit)` should return `0` for `si_offset` for quantities other than temperature, so that the condition below is not needed - the value is always given by `(_value + si_offset) * si_multiplier`.

Reply:
Removed hard-coded check for temperature. Now if the unit string is a temperature it gets the offset, anything else the offset is set to 0.

https://github.com/ansys/pyfluent/blob/9cbf840c81b3652466d7c61e22f63a5f0744907d/src/ansys/fluent/core/quantity/quantity.py#L57
https://github.com/ansys/pyfluent/blob/9cbf840c81b3652466d7c61e22f63a5f0744907d/src/ansys/fluent/core/quantity/quantity.py#L92
https://github.com/ansys/pyfluent/blob/9cbf840c81b3652466d7c61e22f63a5f0744907d/src/ansys/fluent/core/quantity/quantity.py#L186
https://github.com/ansys/pyfluent/blob/9cbf840c81b3652466d7c61e22f63a5f0744907d/src/ansys/fluent/core/quantity/units_table.py#L173-L177